numpyro/primitives.py

```diff
@@ -649,6 +650,105 @@ def plate_stack(
         yield
 
 
+class label_event_dim(plate):
```

Comment:
might be good to break the dependence on `plate` and inherit Messenger instead - this was a quick hack to get things working